src/stravalib/client.py

```diff
@@ -451,26 +451,30 @@ def get_athlete_stats(
 
         return model.AthleteStats.model_validate(raw)
 
-    def get_athlete_clubs(self) -> list[model.Club]:
+    def get_athlete_clubs(
+        self, limit: int | None = None
+    ) -> BatchedResultsIterator[model.SummaryClub]:
         """List the clubs for the currently authenticated athlete.
 
         https://developers.strava.com/docs/reference/#api-Clubs-getLoggedInAthleteClubs
 
+        limit : int or None, default=None
+            Maximum number of club objects to return.
 
         Returns
         -------
-        :class:`list`
-            A list of :class:`stravalib.model.Club`
+        class:`BatchedResultsIterator`
+            An iterator of :class:`stravalib.model.SummaryClub` objects
 
         """
 
-        # TODO: This should return a BatchedResultsIterator or otherwise at
-        # most 30 clubs are returned!
-        club_structs = self.protocol.get("/athlete/clubs")
-        return [
-            model.Club.model_validate({**raw, **{"bound_client": self}})
-            for raw in club_structs
-        ]
+        result_fetcher = functools.partial(self.protocol.get, "/athlete/clubs")
```

Comment:
do we need params to specify per page, etc? i wasn't sure. i only have 3 clubs and one is a dummy club that i created. so i can't test on that many clubs.

Reply:
Do you mean we maybe need to pass additional arguments to `functools.partial()`?

Reply:
@jsamoocha - sorry! your interpretation is correct. Do we need to pass `functools.partial()` any additional arguments? I was worried about this because other methods do get other arguments for instance get_club_members ingests club_id in the id param. and get activities gets the params of before and after, etc. 

I think the answer here is no given this just returns the authenticated athlete's clubs so there might be nothing to pass. But i wanted to be certain there wasnt something like a param to specify how many clubs to return that i was missing.